etcd panic with long-running compact request

I have been running this code https://github.com/coreos/etcd/pull/5282 with testing cluster and got this server side panic:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x4a916f]

goroutine 412563 [running]:
panic(0xb53180, 0xc82000e0c0)
	/home/gyuho/go-master/src/runtime/panic.go:500 +0x18c
github.com/coreos/etcd/cmd/vendor/github.com/coreos/etcd/etcdserver.(*EtcdServer).Compact(0xc82014c200, 0x102cac0, 0xc83aa531d0, 0xc830643f80, 0x61bec9, 0xc820189760, 0x102da00)
	/home/gyuho/go/src/github.com/coreos/etcd/cmd/vendor/github.com/coreos/etcd/etcdserver/v3_server.go:124 +0xbf
github.com/coreos/etcd/cmd/vendor/github.com/coreos/etcd/etcdserver/api/v3rpc.(*kvServer).Compact(0xc83dbb8a80, 0x102cac0, 0xc83aa531d0, 0xc830643f80, 0x102f800, 0x3e1c5c00, 0xbd4ea0)
	/home/gyuho/go/src/github.com/coreos/etcd/cmd/vendor/github.com/coreos/etcd/etcdserver/api/v3rpc/key.go:114 +0x57
github.com/coreos/etcd/cmd/vendor/github.com/coreos/etcd/etcdserver/api/v3rpc.(*quotaKVServer).Compact(0xc83dbb8ac0, 0x102cac0, 0xc83aa531d0, 0xc830643f80, 0x572cfba1, 0xc94a031, 0x5c9588)
	<autogenerated>:7 +0x5d
github.com/coreos/etcd/cmd/vendor/github.com/coreos/etcd/etcdserver/etcdserverpb._KV_Compact_Handler.func1(0x102cac0, 0xc83aa531d0, 0xbc68c0, 0xc830643f80, 0x102f800, 0xc83e1c5c00, 0xc830643e88, 0x18)
	/home/gyuho/go/src/github.com/coreos/etcd/cmd/vendor/github.com/coreos/etcd/etcdserver/etcdserverpb/rpc.pb.go:2159 +0x98
github.com/coreos/etcd/cmd/vendor/github.com/coreos/etcd/etcdserver/api/v3rpc.metricsUnaryInterceptor(0x102cac0, 0xc83aa531d0, 0xbc68c0, 0xc830643f80, 0xc834a43300, 0xc834a43320, 0x7, 0x0, 0x0, 0x44fa70)
	/home/gyuho/go/src/github.com/coreos/etcd/cmd/vendor/github.com/coreos/etcd/etcdserver/api/v3rpc/metrics.go:59 +0x176
github.com/coreos/etcd/cmd/vendor/github.com/coreos/etcd/etcdserver/etcdserverpb._KV_Compact_Handler(0xb92240, 0xc83dbb8ac0, 0x102cac0, 0xc83aa531d0, 0xc83e26b580, 0xcb1d18, 0x0, 0xc59785, 0xe, 0x0)
	/home/gyuho/go/src/github.com/coreos/etcd/cmd/vendor/github.com/coreos/etcd/etcdserver/etcdserverpb/rpc.pb.go:2161 +0x159
github.com/coreos/etcd/cmd/vendor/google.golang.org/grpc.(*Server).processUnaryRPC(0xc827f64900, 0x102efc0, 0xc821230cf0, 0xc8324300e0, 0xc826ec1640, 0x104bec0, 0xc83aa531a0, 0x0, 0x0)
	/home/gyuho/go/src/github.com/coreos/etcd/cmd/vendor/google.golang.org/grpc/server.go:522 +0x947
github.com/coreos/etcd/cmd/vendor/google.golang.org/grpc.(*Server).handleStream(0xc827f64900, 0x102efc0, 0xc821230cf0, 0xc8324300e0, 0xc83aa531a0)
	/home/gyuho/go/src/github.com/coreos/etcd/cmd/vendor/google.golang.org/grpc/server.go:682 +0x68a
github.com/coreos/etcd/cmd/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc830643e20, 0xc827f64900, 0x102efc0, 0xc821230cf0, 0xc8324300e0)
	/home/gyuho/go/src/github.com/coreos/etcd/cmd/vendor/google.golang.org/grpc/server.go:348 +0x9f
created by github.com/coreos/etcd/cmd/vendor/google.golang.org/grpc.(*Server).serveStreams.func1
	/home/gyuho/go/src/github.com/coreos/etcd/cmd/vendor/google.golang.org/grpc/server.go:349 +0x97

```


https://storage.googleapis.com/etcd-issues-5281/tester.log

```
2016-05-06 20:15:54.057002 I | etcd-tester: [round#29 case#10] starting failure slow down all members' network
2016-05-06 20:15:54.057048 I | etcd-tester: [round#29 case#10] injecting failure...
2016-05-06 20:15:55.449601 I | etcd-tester: [round#29 case#10] injected failure
2016-05-06 20:15:55.449634 I | etcd-tester: [round#29 case#10] recovering failure...
2016-05-06 20:16:02.075344 I | etcd-tester: [round#29 case#10] recovered failure
2016-05-06 20:16:02.075401 I | etcd-tester: [round#29 case#10] updating current revisions...
2016-05-06 20:16:03.212435 I | etcd-tester: [round#29 case#10] #0 inconsistent current revisions map[10.240.0.17:2379:67389887 10.240.0.18:2379:67391308 10.240.0.19:2379:67395221]
2016-05-06 20:16:03.212470 I | etcd-tester: [round#29 case#10] updated current revisions with 67395221
2016-05-06 20:16:03.212477 I | etcd-tester: [round#29 case#10] succeed!
2016-05-06 20:16:03.212491 I | etcd-tester: [round#29] compacting storage at 67385221 (current revision 67395221)
2016-05-06 20:16:33.212860 I | etcd-tester: [compactKV #0] kvc.Compact 10.240.0.17:2379 error rpc error: code = 4 desc = context deadline exceeded
2016-05-06 20:16:42.846420 I | etcd-tester: [compactKV #1] kvc.Compact 10.240.0.17:2379 error rpc error: code = 13 desc = transport is closing
2016-05-06 20:16:47.441283 I | etcd-tester: [compactKV #2] kvc.Compact 10.240.0.17:2379 error rpc error: code = 2 desc = grpc: the client connection is closing
2016-05-06 20:16:50.029374 I | etcd-tester: [compactKV #3] kvc.Compact 10.240.0.17:2379 error rpc error: code = 2 desc = grpc: the client connection is closing
2016-05-06 20:16:52.677377 I | etcd-tester: [compactKV #4] kvc.Compact 10.240.0.17:2379 error rpc error: code = 2 desc = grpc: the client connection is closing
2016-05-06 20:17:02.773428 I | etcd-tester: 10.240.0.18:2379 is already compacted with 67385221 (rpc error: code = 11 desc = etcdserver: mvcc: required revision has been compacted)
2016-05-06 20:17:02.848322 I | etcd-tester: 10.240.0.19:2379 is already compacted with 67385221 (rpc error: code = 11 desc = etcdserver: mvcc: required revision has been compacted)
2016-05-06 20:17:02.848366 I | etcd-tester: [round#29] compacted storage at 67385221
2016-05-06 20:17:02.848375 I | etcd-tester: [round#29] checking compaction at 67385221
2016-05-06 20:17:05.765212 I | etcd-tester: [round#29] checkCompact error (grpc: timed out trying to connect)
2016-05-06 20:17:05.765263 I | etcd-tester: [round#29] cleaning up...
2016-05-06 20:17:07.127163 W | etcd-tester: [round#29] functional-tester returning with error (grpc: timed out trying to connect)
```

--- a/etcdserver/v3_server.go
+++ b/etcdserver/v3_server.go
@@ -121,7 +121,7 @@ func isTxnSerializable(r *pb.TxnRequest) bool {
 
 func (s *EtcdServer) Compact(ctx context.Context, r *pb.CompactionRequest) (*pb.CompactionResponse, error) {
 	result, err := s.processInternalRaftRequest(ctx, pb.InternalRaftRequest{Compaction: r})
-	if r.Physical && result.physc != nil {
+	if r.Physical && result != nil && result.physc != nil {
 		<-result.physc
 		// The compaction is done deleting keys; the hash is now settled
 		// but the data is not necessarily committed. If there's a crash,

